### PR TITLE
Create user

### PR DESCRIPTION
**Pull request by vfilimo**

### CODE REVIEW

File: src/main/java/online/store/book/service/user/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        if (userRepository.findUserByEmail(requestUser.getEmail()).isPresent()) {
+            throw new RegistrationException("User with email: "
+                    + requestUser.getEmail() + " exists");
+        }
+        User user = new User();
+        user.setEmail(requestUser.getEmail());
+        user.setPassword(requestUser.getPassword());
+        user.setFirstName(requestUser.getFirstName());
+        user.setLastName(requestUser.getLastName());
+        user.setShippingAddress(requestUser.getShippingAddress());
```
Comment (okuzan) [inline src/main/java/online/store/book/service/user/UserServiceImpl.java]: looks like a job for a mapper!

**Review by okuzan — CHANGES_REQUESTED** [review]
let's rename annotation to validation or constraing package for clarity

File: src/main/java/online/store/book/service/user/UserService.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package online.store.book.service.user;
+
+import online.store.book.dto.user.UserRegistrationRequestDto;
+import online.store.book.dto.user.UserResponseDto;
+import online.store.book.exceptions.RegistrationException;
+
+public interface UserService {
+    public UserResponseDto register(UserRegistrationRequestDto requestUser)
```
Comment (Elena-Bruyako) [inline src/main/java/online/store/book/service/user/UserService.java]:
```suggestion
    UserResponseDto register(UserRegistrationRequestDto requestUser)
```

File: src/main/java/online/store/book/service/user/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestUser)
+            throws RegistrationException {
+        if (userRepository.findUserByEmail(requestUser.getEmail()).isPresent()) {
```
Comment (Elena-Bruyako) [inline src/main/java/online/store/book/service/user/UserServiceImpl.java]:
```suggestion
        if (userRepository.existsByEmail(requestUser.getEmail())) {
```

File: src/main/java/online/store/book/validation/FieldMatchValidator.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        this.firstFieldName = constraintAnnotation.first();
+        this.secondFieldName = constraintAnnotation.second();
+    }
+
+    @Override
+    public boolean isValid(Object obj, ConstraintValidatorContext constraintValidatorContext) {
+        try {
```
Comment (Elena-Bruyako) [inline src/main/java/online/store/book/validation/FieldMatchValidator.java]: Or you can use 

Object field = new BeanWrapperImpl(value).getPropertyValue(this.firstFieldName;
Object fieldMatch = new BeanWrapperImpl(value).getPropertyValue(this.secondFieldName);
return Objects.equals(field, fieldMatch);

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve it

**Review by Elena-Bruyako — APPROVED** [review]
well done

**Pull request merged**